src/command_modules/azure-cli-eventgrid/azure/cli/command_modules/eventgrid/custom.py

```diff
@@ -108,82 +188,77 @@ def cli_eventgrid_event_subscription_get(
-        # Resource ID is specified, we need to list only for the particular resource.
-        if resource_group_name is not None or topic_name is not None:
-            raise CLIError('Since ResourceId is specified, topic-name and resource-group-name should not be specified.')
-
-        id_parts = parse_resource_id(resource_id)
-        rg_name = id_parts['resource_group']
-        resource_name = id_parts['name']
-        provider_namespace = id_parts['namespace']
-        resource_type = id_parts['resource_type']
-
-        return client.list_by_resource(
-            rg_name,
-            provider_namespace,
-            resource_type,
-            resource_name)
+    if source_resource_id is not None:
+        # If Source Resource ID is specified, we need to list event subscriptions for that particular resource.
+        # Since a full resource ID is specified, it should override all other defaults such as default location and RG
+        # No other parameters must be specified
+        if (topic_type_name is not None or resource_id is not None):
+            raise CLIError('usage error: Since --source-resource-id is specified, none of the other parameters must '
+                           'be specified.')
+
+        return _list_event_subscriptions_by_resource_id(client, source_resource_id)
+
+    if resource_id is not None:
+        # DEPRECATED
+        # If resource ID is specified, we need to list event subscriptions for that particular resource.
+        # Since a full resource ID is specified, it should override all other defaults such as default location and RG
+        # No other parameters must be specified
+        if topic_type_name is not None:
+            raise CLIError('usage error: Since --resource-id is specified, none of the other parameters must '
+                           'be specified.')
+
+        return _list_event_subscriptions_by_resource_id(client, resource_id)
 
     if topic_name:
+        # DEPRECATED
         if resource_group_name is None:
-            raise CLIError('Since topic-name is specified, resource-group-name must also be specified.')
+            raise CLIError('Since --topic-name is specified, --resource-group must also be specified.')
 
         return client.list_by_resource(
             resource_group_name,
             EVENTGRID_NAMESPACE,
             EVENTGRID_TOPICS,
             topic_name)
 
-    if topic_type_name:
-        if location:
-            if resource_group_name:
-                return client.list_regional_by_resource_group_for_topic_type(
-                    resource_group_name,
-                    location,
-                    topic_type_name)
+    if location is None:
+        # Since resource-id was not specified, location must be specified: e.g. "westus2" or "global". If not error OUT.
+        raise CLIError('usage error: --source-resource-id ID | --location LOCATION'
+                       ' [--resource-group RG] [--topic-type-name TOPIC_TYPE_NAME]')
```

Comment:
I would highly recommend doing this validation in a [validator](https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#validators).

Reply:
Thanks for the suggestion. We need to keep this code as is here since it is related with the previous code context. This validation is only used when resource-id is not specified. I rather keep it this way for better code readability.